homeassistant/components/homematic/sensor.py

```diff
@@ -45,54 +59,174 @@
-    "GAS_POWER": VOLUME_CUBIC_METERS,
-    "GAS_ENERGY_COUNTER": VOLUME_CUBIC_METERS,
-    "IEC_POWER": POWER_WATT,
-    "IEC_ENERGY_COUNTER": ENERGY_WATT_HOUR,
-    "LUX": LIGHT_LUX,
-    "ILLUMINATION": LIGHT_LUX,
-    "CURRENT_ILLUMINATION": LIGHT_LUX,
-    "AVERAGE_ILLUMINATION": LIGHT_LUX,
-    "LOWEST_ILLUMINATION": LIGHT_LUX,
-    "HIGHEST_ILLUMINATION": LIGHT_LUX,
-    "RAIN_COUNTER": LENGTH_MILLIMETERS,
-    "WIND_SPEED": SPEED_KILOMETERS_PER_HOUR,
-    "WIND_DIRECTION": DEGREE,
-    "WIND_DIRECTION_RANGE": DEGREE,
-    "SUNSHINEDURATION": "#",
-    "AIR_PRESSURE": PRESSURE_HPA,
-    "FREQUENCY": FREQUENCY_HERTZ,
-    "VALUE": "#",
-    "VALVE_STATE": PERCENTAGE,
-    "CARRIER_SENSE_LEVEL": PERCENTAGE,
-    "DUTY_CYCLE_LEVEL": PERCENTAGE,
-    "CONCENTRATION": CONCENTRATION_PARTS_PER_MILLION,
-}
 
-HM_DEVICE_CLASS_HA_CAST = {
-    "HUMIDITY": DEVICE_CLASS_HUMIDITY,
-    "TEMPERATURE": DEVICE_CLASS_TEMPERATURE,
-    "ACTUAL_TEMPERATURE": DEVICE_CLASS_TEMPERATURE,
-    "LUX": DEVICE_CLASS_ILLUMINANCE,
-    "CURRENT_ILLUMINATION": DEVICE_CLASS_ILLUMINANCE,
-    "AVERAGE_ILLUMINATION": DEVICE_CLASS_ILLUMINANCE,
-    "LOWEST_ILLUMINATION": DEVICE_CLASS_ILLUMINANCE,
-    "HIGHEST_ILLUMINATION": DEVICE_CLASS_ILLUMINANCE,
-    "POWER": DEVICE_CLASS_POWER,
-    "CURRENT": DEVICE_CLASS_POWER,
-    "CONCENTRATION": DEVICE_CLASS_CO2,
+SENSOR_DESCRIPTIONS: dict[str, SensorEntityDescription] = {
+    "HUMIDITY": SensorEntityDescription(
+        key="HUMIDITY",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=DEVICE_CLASS_HUMIDITY,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    "ACTUAL_TEMPERATURE": SensorEntityDescription(
+        key="ACTUAL_TEMPERATURE",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    "TEMPERATURE": SensorEntityDescription(
+        key="TEMPERATURE",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    "LUX": SensorEntityDescription(
+        key="LUX",
+        native_unit_of_measurement=LIGHT_LUX,
+        device_class=DEVICE_CLASS_ILLUMINANCE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    "CURRENT_ILLUMINATION": SensorEntityDescription(
+        key="CURRENT_ILLUMINATION",
+        native_unit_of_measurement=LIGHT_LUX,
+        device_class=DEVICE_CLASS_ILLUMINANCE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    "ILLUMINATION": SensorEntityDescription(
+        key="ILLUMINATION",
+        native_unit_of_measurement=LIGHT_LUX,
+        device_class=DEVICE_CLASS_ILLUMINANCE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    "AVERAGE_ILLUMINATION": SensorEntityDescription(
+        key="AVERAGE_ILLUMINATION",
+        native_unit_of_measurement=LIGHT_LUX,
+        device_class=DEVICE_CLASS_ILLUMINANCE,
+        state_class=STATE_CLASS_TOTAL,
```

Comment:
Are you sure `STATE_CLASS_TOTAL` is correct for this one and the next two? Shouldn't it be `STATE_CLASS_MEASUREMENT`?

Reply:
Hmm, I think you are right. I read the documentation again and it says:

> https://developers.home-assistant.io/docs/core/entity/sensor/#available-state-classes
This state class should not be used for sensors where the absolute value is interesting instead of the accumulated growth or decline, for example remaining battery capacity or CPU load; in such cases state class measurement should be used instead.

Sometimes it's really hard to tell, which one to use. Thanks for the hint :+1: 
I will change that.